theme/views.py

```diff
@@ -30,11 +30,21 @@ def get_context_data(self, **kwargs):
             except:
                 u = User.objects.get(username=self.request.GET['user'])
 
-        res = BaseResource.objects.filter(user=u)
+        # get all resources the profile user has created
+        resources = BaseResource.objects.filter(user=u)
+
+        # if requesting user is not the profile user, then show only resources that the requesting user has access
+        if self.request.user != u:
+            if self.request.user.is_authenticated():
+                # filter out any resources the requesting user doesn't have view permission
+                resources = [res for res in resources if self.request.user.uaccess.can_view_resource(res)]
+            else:
+                # for anonymous requesting user show only resources that are either public or discoverable
+                resources = [res for res in resources if res.raccess.public or res.raccess.discoverable]
```

Comment:
@pkdash @alvacouch Looking at the first line of the code snippet above for can_view_resource(), do you think the line should be changed to "if access_resource.public or access_resource.discoverable" to account for discoverable resources as well?